ravenframework/Optimizers/BayesianOptimizer.py

```diff
@@ -0,0 +1,758 @@
+    """
+    result = self._model.supervisedContainer[0].model.log_marginal_likelihood(logTheta, eval_gradient=True, clone_kernel=False)
+    # Flipping sign for maximization
+    logLikelihood = -1 * result[0]
+    lmlGradient = -1 * result[1]
+    return logLikelihood, lmlGradient
+
+  def _selectHyperparameters(self, restartCount=5):
+    """
+      Selects MAP model in model space, and is exclusively for sklearn GPR models
+      @ In, restartCount, int, number of initial points to start optimization from
+      @ Out, None
+    """
+    # Build bounds for optimization
+    paramBounds = []
+
+    for hyperParam in self._model.supervisedContainer[0].model.kernel.hyperparameters:
+      for bound in hyperParam.bounds:
+        paramBounds.append(tuple(np.log(bound)))
+
+    # Restart locations include current parameter values
+    sampler = LHS(xlimits=np.array(paramBounds), criterion='cm', random_state=self._seed)
+    initSamples = sampler(restartCount-1)
+    currentTheta = np.array([self._model.supervisedContainer[0].model.kernel.theta])
+    initSamples = np.concatenate((initSamples, currentTheta))
+    # Selecting MAP for each restart
+    options = {'ftol':1e-10, 'maxiter':200, 'disp':False}
+    res = None
+
+    for x0 in initSamples:
+      result = sciopt.minimize(self._logMarginalLikelihood, x0, method='SLSQP', jac=True, bounds=paramBounds, options=options)
+      if res is None:
+        res = result
+      elif result.fun < res.fun:
+        res = result
+    newTheta = res.x
+    newKernel = self._model.supervisedContainer[0].model.kernel_.clone_with_theta(newTheta)
+    self._model.supervisedContainer[0].model.kernel = newKernel
+    self._trainRegressionModel(0)
+
+  def _trainRegressionModel(self, traj):
+    """
+      Reformats training data into form that ROM can handle
+      @ In, traj, trajectory for training the model
+      @ Out, None
+    """
+    # Build training set to feed to rom model
+    trainingSet = {}
+
+    for varName in list(self.toBeSampled):
+      trainingSet[varName] = np.asarray(self._trainingInputs[traj][varName])
+    trainingSet[self._objectiveVar] = np.asarray(self._trainingTargets[traj])
+    self._model.train(trainingSet)
+    # NOTE It would be preferrable to use targetEvaluation;
+    # however, there does not appear a built in normalization method and as
+    # consequence, it is preferrable to use the in class attributes to train the ROM
+    # on our normalized space.
+    # self._model.train(self._targetEvaluation)
+
+  def _generatePredictiveModel(self, traj):
+    """
+      Applies model selection if necessary, otherwise just fits training data to predictive model
+      @ In, traj, trajectory
+      @ Out, None
+    """
+    # Generate posterior with training data
+    if self._iteration[traj] % self._modelDuration == 0:
+      if self._modelSelection == 'External':
+        self._model.supervisedContainer[0].model.set_params(optimizer='fmin_l_bfgs_b')
```

Comment:
General suggestion: I prefer to define a self variable to store the hard-coded value.

Reply:
I am not sure what in these lines you are referring to.

Reply:
I mean the value that you are used in optimizer. In the places, such as tolerance, max iteration, etc. Instead of directly assign the value, I recommend to collect them and put them inside class __init__ function.